scripts/build_pdf.py

```
@@ -0,0 +1,231 @@
+        print(f"  ✓ Found {len(self.chapters)} chapters")
+        return len(self.chapters) > 0
+
+    def compile_pdf(self) -> bool:
+        """Compile all chapters into single PDF using Playwright."""
+
+        if not PLAYWRIGHT_AVAILABLE:
+            print("ERROR: Playwright is required for PDF compilation")
+            print("Install: pip install playwright && playwright install chromium")
+            return False
+
+        print(f"\n[1/3] Loading chapters...")
```

Ruff (0.14.4)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)